adafruit_pyportal.py

```diff
@@ -303,34 +310,39 @@ def __init__(self, *, url=None, json_path=None, regexp_path=None,
         # pylint: enable=no-member
 
         self.set_backlight(1.0)  # turn on backlight
+
         gc.collect()
 
-    def set_background(self, filename):
+    def set_background(self, file_or_color):
         """The background image to a bitmap file.
 
-        :param filename: The name of the chosen background image file.
+        :param file_or_color: The filename of the chosen background image, or a hex color.
 
         """
-        print("Set background to ", filename)
-        try:
+        print("Set background to ", file_or_color)
+        while len(self._bg_group) != 0:
```

Comment:
I think this can be `while len(self._bg_group):` but I have no way to test it. If it is `= 0` then it is false. So in theory, if it's `!= 0` it's true.

Reply:
hiya im still making changes, and will do all the lints once things work

Reply:
Got it!